repair_scheduled_date_calendar_view/models/repair_order.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2025 Cetmix OÜ
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class RepairOrder(models.Model):
+    _inherit = "repair.order"
+
+    # Magic values injected by Odoo calendar widget (frontend)
+    CALENDAR_DEFAULT_DURATION = 1.0  # injected when creating a new event
+    CALENDAR_ALLDAY_DURATION = 12.0  # injected when creating an all-day event
+
+    def _get_default_planned_duration(self):
+        """Get default planned duration from system configuration."""
+        return float(
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param(
+                "repair_scheduled_date_calendar_view.planned_duration_default",
+                1.0,
+            )
+        )
+
+    planned_duration = fields.Float(
+        default=lambda self: self._get_default_planned_duration(),
+        help="Planned duration of the repair order in hours.",
+    )
+
+    @api.model
+    def default_get(self, fields_list):
+        res = super().default_get(fields_list)
+
+        if "planned_duration" in fields_list:
+            ctx_val = self.env.context.get("default_planned_duration")
+            # Calendar widget forces either 1.0 or 12.0 by default
+            if ctx_val in (
+                self.CALENDAR_DEFAULT_DURATION,
+                self.CALENDAR_ALLDAY_DURATION,
+            ):
+                res["planned_duration"] = self._get_default_planned_duration()
+            elif res.get("planned_duration") is None:
```

Comment:
Suppose someone would like to create a 1 hour repair order from another model and add "default_duration: 1.0". This function will not allow to do this.